### PR TITLE
Configurable http api prefix

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
- Adds a configurable http api endpoint prefix
- Resets the http query endpoint to be rooted at `/api` instead of `/tempo/api`

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by annanay25 — COMMENTED** [review]
Nice. We will also need changes to the jsonnet/mixin.

**Review by mdisibio — APPROVED** [review]

**Comment by joe-elliott** [comment]
@annanay25 mixin fixed.

are there places in the deployment jsonnet that need updating?

**Comment by annanay25** [comment]
Nope, I think this is all.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
   This is a **breaking change** and will likely result in query errors on rollout as the query signature b/n QueryFrontend & Querier has changed. [#557](https://github.com/grafana/tempo/pull/557)
 * [ENHANCEMENT] Add list compaction-summary command to tempo-cli [#588](https://github.com/grafana/tempo/pull/588)
 * [ENHANCEMENT] Add list and view index commands to tempo-cli [#611](https://github.com/grafana/tempo/pull/611)
+* [ENHANCEMENT] Add a configurable prefix for HTTP endpoints. [#631](https://github.com/grafana/tempo/pull/631)
```
Comment (annanay25) [inline CHANGELOG.md:15]: Does this count as a breaking change?

**Review by annanay25 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (joe-elliott) [reply CHANGELOG.md:15]: Unsure. It technically unbreaks the hardcoded `/tempo` breaking change :). So 0.6.0 => 0.7.0 will not be a breaking change.  While this exact commit is a breaking change from the previous one.

I think for this reason we leave it out.

**Pull request merged**